From 6d17bd23274e99aa5f1124dc1af43263d880db7b Mon Sep 17 00:00:00 2001
From: sakindia123 <sa6915@gmail.com>
Date: Mon, 15 Jul 2013 02:25:56 +0100
Subject: [PATCH] Makefile Optimizations

---
 Makefile | 26 ++++++++++++++++++++++----
 1 file changed, 22 insertions(+), 4 deletions(-)
 mode change 100755 => 100644 Makefile

diff --git a/Makefile b/Makefile
old mode 100755
new mode 100644
index c841f06..002a33a
--- a/Makefile
+++ b/Makefile
@@ -347,16 +347,27 @@ KALLSYMS	= scripts/kallsyms
 PERL		= perl
 CHECK		= sparse
 
+ARM_FLAGS       = -funswitch-loops \
+                  -fpredictive-commoning \
+                  -fgcse-after-reload \
+                  -fipa-cp-clone \
+                  -fsingle-precision-constant \
+                  -pipe -finline-functions \
+                  -ffast-math \
+                  -mfpu=neon \
+                  -march=armv7-a \
+                  -fvect-cost-model
+
 # Use the wrapper for the compiler.  This wrapper scans for new
 # warnings and causes the build to stop upon encountering them.
 CC		= $(srctree)/scripts/gcc-wrapper.py $(REAL_CC)
 
 CHECKFLAGS     := -D__linux__ -Dlinux -D__STDC__ -Dunix -D__unix__ \
 		  -Wbitwise -Wno-return-void $(CF)
-CFLAGS_MODULE   =
-AFLAGS_MODULE   =
-LDFLAGS_MODULE  =
-CFLAGS_KERNEL	= -march=armv7-a -mfpu=neon -ftree-vectorize -ffast-math -fsingle-precision-constant
+CFLAGS_MODULE   = $(ARM_FLAGS) -DMODULE
+AFLAGS_MODULE   = $(ARM_FLAGS) -DMODULE --strip-debug
+LDFLAGS_MODULE  = -T $(srctree)/scripts/module-common.lds
+CFLAGS_KERNEL	= $(ARM_FLAGS)
 AFLAGS_KERNEL	= 
 CFLAGS_GCOV	= -fprofile-arcs -ftest-coverage
 
@@ -373,7 +384,14 @@ KBUILD_CPPFLAGS := -D__KERNEL__
 KBUILD_CFLAGS   := -Wno-trigraphs \
 		   -fno-strict-aliasing -fno-common \
 		   -Wno-format-security \
+                   -marm -mfloat-abi=softfp -march=armv7-a \
+                   -mfpu=neon -ffast-math -pipe \
+                   -funswitch-loops -fpredictive-commoning -fgcse-after-reload -fno-tree-vectorize \
+                   -ftree-vectorize -funsafe-math-optimizations \
+                   -fsched-spec-load -mvectorize-with-neon-quad \
+                   -fmodulo-sched -fmodulo-sched-allow-regmoves \
 		   -fno-delete-null-pointer-checks
+
 KBUILD_AFLAGS_KERNEL :=
 KBUILD_CFLAGS_KERNEL :=
 KBUILD_AFLAGS   := -D__ASSEMBLY__
-- 
1.8.1.6
